### ISSUE TITLE
Modification summary per BED line

### ISSUE THREAD

**Issue by davidsilvapires** [nanoporetech/modkit#154]
Hi everybody.

We have a bed file with about 1000 features (one per line) that we are interested in obtaining the number of modified bases per feature (normalized to coverage, preferentially).

Is there any modkit sub-command similar to "summary" to get this kind of result?

Thanks in advance for any help.

--
David

**Comment by ArtRand** [comment]
Hello @davidsilvapires,

There isn't a command exactly like that. Although I've gotten a similar request before - so maybe it's time to add one.

**Comment by davidsilvapires** [comment]
Thank you for the answer @ArtRand.

While the subcommand is not implemented, I am using the <() Bash syntax to make use of modkit summary running one BED line each time. Maybe it can useful to other users:

```bash
IFS='
'

for LINE in `cat features.bed`
do
     echo
     echo "${LINE}"
     modkit summary --include-bed <(echo ${LINE}) sample.modBam
done
```

Here are two output examples:
                                                                               
PRFA01000001    1129695 1129767 C4B63_1nc1      tRNA    -                                                                                                                                                                                                                                                                   
# bases             C                                                                                                                                                                                                                                                                                                       
# total_reads_used  17                                                                                                                                                                                                                                                                                                      
# count_reads_C     17                                                                                                                                                                                                                                                                                                      
# pass_threshold_C  0.6507813                                                  
 base  code  pass_count  pass_frac   all_count  all_frac                                                                                                                                                                                                                                                                    
 C     -     91          0.9009901   95         0.84070796                                                                                                                                                                                                                                                                  
 C     m     4           0.03960396  6          0.053097345                                                                                                   
 C     h     6           0.05940594  12         0.10619469                                                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                                                                                                            
PRFA01000005    770166  770238  C4B63_5nc4      tRNA    -                                                                                                                                                                                                                                                                   
# bases             C                                                                                                                                                                                                                                                                                                       
# total_reads_used  12                                                                                                                                                                                                                                                                                                      
# count_reads_C     12                                                                                                                                                                                                                                                                                                      
# pass_threshold_C  0.74921876                                                                                                                                                                                                                                                                                              
 base  code  pass_count  pass_frac   all_count  all_frac                                                                                                                                                                                                                                                                    
 C     -     70          0.93333334  76         0.9047619                      
 C     h     5           0.06666667  7          0.083333336                                                                                                                                                                                                                                                                  C     m     0           0           0          0                                                                                                                                                                                                                                                                                                                                                                                                                                         


Best regards.

--
David

**Comment by ArtRand** [comment]
Hello @davidsilvapires, 

A feature like this called `modkit stats` is available in [v0.4.1](https://github.com/nanoporetech/modkit/releases/tag/v0.4.1) (actually it was released in v0.4.0). Don't hesitate to open a separate issue if you have any questions with that command.

**Comment by Soniazumalave** [comment]
Hello!

Thank you for implementing this—it's very useful. Would it be possible to add an argument like --cpg in this subcommand to only consider CpG sites?
